src/ReTestItems.jl

```diff
@@ -396,12 +439,11 @@ function record_test_error!(testitem, msg, elapsed_seconds::Real=0.0)
     return testitem
 end
 
-function start_and_manage_worker(
-    proj_name, testitems, testitem, nworker_threads, worker_init_expr,
+function manage_worker(
+    worker::Worker, proj_name, testitems, testitem, nworker_threads, worker_init_expr,
     timeout::Real, retries::Int, verbose_results::Bool, debug::Int, report::Bool, logs::Symbol
 )
     ntestitems = length(testitems.testitems)
-    worker = start_worker(proj_name, nworker_threads, worker_init_expr, ntestitems)
```

Comment:
Any thoughts about the alternative design where we keep spawning the workers eagerly here, but with `robust_start_worker` so they can start working asap? If any of the workers failed to start at this point, we would throw an exception which would be caught in the caller, `terminate!`ing all the other workers and failing the run (so we'd still require all workers to start successfully for `runtests` to succeed).

Reply:
perhaps i'm not understanding, but i think in that case the runtests would fail _after_ all the successful workers had completed running there tests (i.e. potentially after hours of unnecessary processing)

Reply:
Some workers might be in the middle of running tests, yes, but if we detect that any worker failed to start with retries, we'd call `terminate!` on _all_ workers, so they should be interrupted/killed quickly (like we do when they time-out).

Reply:
> how would we do that?
I guess with a flag that says whether to retry workers or not, but I didn't mean to delve into implementation detail, I wanted to discuss the overall idea. I think in a lot of cases the latency diff won't be noticeable, it might be depending on what `worker_init_expr` is and how many workers one uses. But if you are set on the current approach I still think it's an improvement, don't get me wrong.

Reply:
i like how simple the current approach here is, and don't _expect_ the latency to be noticeable, which is why i'd rather wait and see if we need to do anything more

Reply:
thanks for making me think about it more!